### DISCUSSION TITLE
Augmentation

### DISCUSSION THREAD

**Discussion by Sarath-C-Koppolu** [ultralytics/ultralytics#13471] [General]
Hello ultralytics do auto augmentation 
I want to know how it works

And if it is doing auto augmentation, i don't have to augment my images right ?

**Comment by glenn-jocher** [comment]
@Sarath-C-Koppolu hello!

Great question! Yes, Ultralytics YOLOv8 does support auto augmentation, which can significantly enhance your model's performance by automatically applying various augmentation techniques to your training data.

Auto augmentation in YOLOv8 leverages predefined policies to apply transformations such as rotation, translation, scaling, and color adjustments to your images. This helps in making the model more robust and generalizable to different scenarios.

Here's a brief overview of how it works:

1. **Predefined Policies**: YOLOv8 uses predefined augmentation policies that apply a series of transformations to your images during training.
2. **Randomization**: These transformations are applied randomly, ensuring that each image is augmented differently, which helps in preventing overfitting.
3. **Integration**: The auto augmentation is seamlessly integrated into the training pipeline, so you don't need to manually augment your images.

To enable auto augmentation, you can simply set the `auto_augment` parameter in your training configuration. Here's an example:

```python
from ultralytics import YOLO

# Load a model
model = YOLO('yolov8n.pt')  # Load a pretrained model

# Train the model with auto augmentation
results = model.train(data='coco128.yaml', epochs=100, imgsz=640, auto_augment='randaugment')
```

In this example, `randaugment` is one of the available auto augmentation policies. You can also explore other policies like `autoaugment` and `augmix`.

To answer your second question, if you enable auto augmentation, you don't need to manually augment your images. The auto augmentation will handle it for you, ensuring a diverse and robust training dataset.

Feel free to experiment with different policies and see which one works best for your specific use case. If you have any more questions or run into any issues, don't hesitate to ask! 😊

Happy training! 🚀
